File: librarymanager/librarymanager-be/src/main/java/ru/elenacherev/librarymanager/controller/BookUsingController.java
package ru.elenacherev.librarymanager.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import ru.elenacherev.librarymanager.api.BookUsingApi;
import ru.elenacherev.librarymanager.api.dto.BookUsing;
import ru.elenacherev.librarymanager.services.BookUsingService;

import javax.validation.Valid;

@RequestMapping(value = "librarymanager/bookUsings", produces = {MediaType.APPLICATION_JSON_VALUE})
@RestController
@RequiredArgsConstructor
@Slf4j
public class BookUsingController implements BookUsingApi {

    BookUsingService bookUsingService;

    @Override
	@GetMapping("/{bookusingid}")
    public BookUsing getBookUsing(
            @PathVariable("bookusingid") Long bookusingid
    ) {
    	log.info("Searching BookUsing by Id {0}", bookusingid );
    	return bookUsingService.findBookUsingById(bookusingid);
    }

    @Override
	@PostMapping("/search")
    public Page<BookUsing> searchBookUsings(
            Pageable pageable
    ) {
        log.info("Searching BookUsings");
        return bookUsingService.findAllByPage(pageable);
    }

    @Override
	@PostMapping
    public BookUsing createBookUsing(
            @Valid @RequestBody BookUsing bookUsing
    ) {
        log.info("Creating BookUsing");
        return bookUsingService.create(bookUsing);
    }

    @Override
	@PutMapping("/{bookusingid}")
    public BookUsing saveBookUsing(
            @PathVariable("bookusingid") Long bookUsingId,
            @Valid @RequestBody BookUsing bookUsing
    ) {
        log.info("Updating bookUsing");
        return bookUsingService.save(bookUsingId, bookUsing);
    }
}

File: librarymanager-be/src/main/java/ru/elenacherev/librarymanager/controller/BookUsingController.kt
package ru.elenacherev.librarymanager.controller

import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.elenacherev.librarymanager.api.BookUsingApi
import ru.elenacherev.librarymanager.api.dto.BookUsing
import ru.elenacherev.librarymanager.services.BookUsingService
import javax.validation.Valid

@RequestMapping("/bookUsings")
@RestController
class BookUsingController(
    val bookUsingService: BookUsingService
) : BookUsingApi {

    @GetMapping("/{bookusingid}")
    override fun getBookUsing(
        @PathVariable("bookusingid") bookusingid: Long
    ) = bookUsingService.findBookUsingById(bookusingid)

    @PostMapping("/search")
    override fun searchBookUsings(
        pageable: Pageable
    ) = bookUsingService.findAllByPage(pageable)

    @PostMapping
    override fun createBookUsing(
        @RequestBody bookUsing: @Valid BookUsing
    ) = bookUsingService.create(bookUsing)

    @PutMapping("/{bookusingid}")
    override fun saveBookUsing(
        @PathVariable("bookusingid") bookUsingId: Long,
        @RequestBody bookUsing: @Valid BookUsing
    ) = bookUsingService.save(bookUsingId, bookUsing)
}